integrations/terraform-cloud/client.py

```diff
@@ -34,40 +36,26 @@ def __init__(self, terraform_base_url: str, auth_token: str) -> None:
         self.client = http_async_client
         self.client.headers.update(self.base_headers)
 
+        self.rate_limiter = AsyncLimiter(NUMBER_OF_REQUESTS, NUMBER_OF_SECONDS)
+
     async def send_api_request(
         self,
         endpoint: str,
         method: str = "GET",
         query_params: Optional[dict[str, Any]] = None,
         json_data: Optional[dict[str, Any]] = None,
     ) -> dict[str, Any]:
-        logger.info(f"Requesting Terraform Cloud data for endpoint: {endpoint}")
-        try:
+        async with self.rate_limiter:
             url = f"{self.api_url}/{endpoint}"
-            logger.info(
-                f"URL: {url}, Method: {method}, Params: {query_params}, Body: {json_data}"
-            )
             response = await self.client.request(
                 method=method,
                 url=url,
                 params=query_params,
                 json=json_data,
             )
             response.raise_for_status()
-
-            logger.info(f"Successfully retrieved data for endpoint: {endpoint}")
```

Comment:
I will prefer we keep this too. and if possible, let's log the pagination object so we see how much we fetched etc 

pagination_meta = response.get("meta", {}).get("pagination", {})